healthcheck.tf

```diff
@@ -0,0 +1,58 @@
+locals {
+  healthchecks = module.this.enabled && var.healthchecks != null ? {
+    for healthcheck in flatten(var.healthchecks) :
+    format("%s-%s-%s",
+      lookup(healthcheck, "name", null) == null ? module.this.id : healthcheck.name,
+      healthcheck.type,
+      healthcheck.address
+    ) => healthcheck
+  } : {}
```

Comment:
As with `firewall_rules`, make the input a `map(any)` rather than a `list(any)` so we can get and use sensible names.